packages/api/src/checkTypes.manual.ts

```diff
@@ -37,6 +37,14 @@ async function query (api: ApiPromise, keyring: TestKeyringMap): Promise<void> {
   const intentions = await api.query.staking.intentions();
   console.log('intentions:', intentions);
 
+  // api.query.*.* is well-typed
+  const bar = await api.query.foo.bar(); // bar is Codec (unknown module)
+  const bal = await api.query.balances.freeBalance(keyring.alice.address); // bal is u128
```

Comment:
The result should be `Balance` - not sure why this defaults to `U128` (Checked it, it is def `U128` as indicated here)

Reply:
Unfortunately i think this is how TS resolves types, if it's a type alias, it'll take the source type.

https://www.typescriptlang.org/play/#code/C4TwDgpgBAglC8UB2BXAtgIwgJwNwCh8ATCAYwBsBDbaAMxSVOAEsB7JKWgCgEoAuWAXyl2AZ2BRKCTr1xQA9PKgADSsqjNRydFmwAaZKwkx8QA

Reply:
One possible solution might to be replace all `type Balance = u128` by `interface Balance extends u128 {}`

Reply:
Actually happy with that - that is actually the way we do it in createType via the registry now, should have put 2 and 2 together here that is is possibly the same issue.